Allow compiler to be specified with CC.

https://github.com/datajerk/c2d/pull/3

Allow CFLAGS and LDFLAGS to be specified.
--- Makefile.orig	2017-05-03 12:43:15.000000000 -0500
+++ Makefile	2023-10-25 22:26:20.000000000 -0500
@@ -1,6 +1,8 @@
 
 WIN32GCC = /usr/local/gcc-4.8.0-qt-4.8.4-for-mingw32/win32-gcc/bin/i586-mingw32-gcc
 
+CFLAGS ?= -O3
+
 all: bin/c2d bin/text2page bin/page2text
 
 windows: bin/c2d.exe bin/text2page.exe bin/page2text.exe
@@ -11,25 +13,25 @@
 	./makeheader
 
 bin/c2d: c2d.c c2d.h
-	gcc -Wall -Wno-missing-braces -I. -O3 -o $@ $< -lm
+	$(CC) -Wall -Wno-missing-braces -I. $(CFLAGS) $(LDFLAGS) -o $@ $< -lm
 
 bin/c2d.exe: c2d.c c2d.h
-	$(WIN32GCC) -Wall -Wno-missing-braces -I. -O3 -o $@ $<
+	$(WIN32GCC) -Wall -Wno-missing-braces -I. $(CFLAGS) $(LDFLAGS) -o $@ $<
 
 bin/text2page: text2page.c
-	gcc -Wall -O3 -o $@ $< -lm
+	$(CC) -Wall $(CFLAGS) $(LDFLAGS) -o $@ $< -lm
 
 bin/text2page.exe: text2page.c
-	$(WIN32GCC) -Wall -O3 -o $@ $<
+	$(WIN32GCC) -Wall $(CFLAGS) $(LDFLAGS) -o $@ $<
 
 bin/page2text: page2text.c
-	gcc -Wall -O3 -o $@ $< -lm
+	$(CC) -Wall $(CFLAGS) $(LDFLAGS) -o $@ $< -lm
 
 bin/page2text.exe: page2text.c
-	$(WIN32GCC) -Wall -O3 -o $@ $<
+	$(WIN32GCC) -Wall $(CFLAGS) $(LDFLAGS) -o $@ $<
 
 bin/mandelbrotgr: mandelbrotgr.c
-	gcc -Wall -O3 -o $@ $< -lm
+	$(CC) -Wall $(CFLAGS) $(LDFLAGS) -o $@ $< -lm
 
 clean:
 	rm -f bin/* *.dsk c2d.h c2d.h.1
